providers/implementations/digests/null_prov.c

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2021 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/crypto.h>
+#include "prov/digestcommon.h"
+#include "prov/implementations.h"
+
+typedef struct {
+    unsigned char nothing;
+} NULLMD_CTX;
+
+static int null_init(NULLMD_CTX *ctx)
+{
+    return 1;
+}
+
+static int null_update(NULLMD_CTX *ctx, const void *data, size_t datalen)
+{
+    return 1;
+}
+
+static int null_final(unsigned char *md, NULLMD_CTX *ctx)
+{
+    return 1;
+}
+
+#undef PROV_FUNC_DIGEST_FINAL
+#define PROV_FUNC_DIGEST_FINAL(name, dgstsize, fin)                            \
```

Comment:
I'm wonder why this custom version of PROV_FUNC_DIGEST_FINAL is necessary. The only difference seems to be the check `outsz >= dgstsize`  which presumably will always be true for the NULL digest since `dgstize` is 0. Or perhaps this triggers compilation warnings? Is that the justification? Possibly a comment explaining that if that is the case.

Reply:
Yes, this causes compilation warnings otherwise.